Vagrantfile

```diff
@@ -229,6 +229,19 @@ Vagrant.configure("2") do |config|
     SCRIPT
   end
 
+  # Let users provide credentials to log in to Docker Hub.
+  if $vconfig['docker_registry_auth']
+
+    config.vm.provision "trigger", :option => "value" do |trigger|
+      trigger.fire do
+        info 'Authenticating with the Docker registry'
+        system "docker -H localhost:2375 login"
```

Comment:
I'm not very clear with using triggers for provisioning.
Where will this run - host or VM?
I assume this need to run one time in the VM to pick up settings from config.json?